packages/components/ContextualMenu/src/SubmenuItem.tsx

```diff
@@ -38,42 +38,52 @@ export const SubmenuItem = compose<SubmenuItemType>({
     // Grabs the context information from Submenu (currently selected menuItem and client's onItemClick callback)
     const context = React.useContext(CMContext);
 
+    const cmRef = useViewCommandFocus(componentRef);
+
+    const onItemHoverIn = React.useCallback(
+      (e) => {
+        userProps.onHoverIn(e);
+      },[]);
+
     const onItemClick = React.useCallback(
       (e) => {
         if (!disabled) {
           context?.onDismissMenu();
-          onClick ? onClick() : context.onItemClick(itemKey);
+          context?.dismissSubmenu && context.dismissSubmenu();
+          onClick ? onClick() : context?.onItemClick(itemKey);
           e.stopPropagation();
         }
       },
       [context, disabled, itemKey, onClick],
     );
 
-    const cmRef = useViewCommandFocus(componentRef);
-
-    const onItemHoverIn = React.useCallback(
-      (e) => {
-        componentRef.current.focus();
-        userProps.onHoverIn(e);
-      },
-      [componentRef],
-    );
-
-    const pressable = useAsPressable({ ...rest, onPress: onItemClick, onHoverIn: onItemHoverIn });
+    const pressable = useAsPressable({ ...rest, onPress: onItemClick, onHoverIn: onItemHoverIn, delayHoverIn: 500 });
 
+    const [submenuItemHovered, setsubmenuItemHovered] = React.useState(false);
+    context.setsubmenuItemHovered = setsubmenuItemHovered;
     // set up state
     const state: SubmenuItemState = {
       ...pressable.state,
       selected: context.selectedKey === userProps.itemKey,
       disabled: userProps.disabled,
       content: !!text,
       icon: !!icon,
+      submenuItemHovered: submenuItemHovered,
     };
 
+    const onMouseEnter = React.useCallback(
+      (e) => {
+        setsubmenuItemHovered(true);
+        pressable.props.onMouseEnter && pressable.props.onMouseEnter(e);
+        e.stopPropagation();
+      },
+      [pressable, setsubmenuItemHovered],
+    );
+
     /*
      * For SubmenuItem, menu is shown on hover.
      */
-    const onKeyUp = useKeyCallback(onItemHoverIn, ' ', 'Enter', 'ArrowRight');
+    const onKeyUp = useKeyCallback(onMouseEnter, ' ', 'Enter', 'ArrowRight');
```

Comment:
Random question, does mouse enter cover pen interactions? I forget.

Reply:
I believe so! Pointer events should inherit mouse events